app/models/hostgroup.rb

```diff
@@ -223,11 +223,12 @@ def clone(name = "")
   end
 
   def hosts_count
-    Host::Managed.authorized(:view_hosts, Host).where(:hostgroup => id).size
+    HostCounter.new(:hostgroup)[self]
   end
 
   def children_hosts_count
-    Host::Managed.authorized.where(:hostgroup => subtree_ids).size
+    counter = HostCounter.new(:hostgroup)
+    subtree_ids.map{|child_id| counter.fetch(child_id, 0)}.sum
```

Comment:
Is is worth caching that as well?

Reply:
I don't think it would be a great save, this is just adding a few numbers that are already in memory (since they are cached in the counter).